pkg/util/kube/namespace_deletionstatus_tracker.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kube
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// NamespaceDeletionStatusTracker keeps track of in-progress backups.
+type NamespaceDeletionStatusTracker interface {
+	// Add informs the tracker that a polling is in progress to check namespace deletion status.
+	Add(ns, name string)
+	// Delete informs the tracker that a namespace deletion is completed.
+	Delete(ns, name string)
+	// Contains returns true if the tracker is tracking the namespace deletion progress.
+	Contains(ns, name string) bool
+}
```

Comment:
Why are there both ns and name parameters?
What purpose does that serve other than duplicate value params?

Is this interface meant to be more generic? Perhaps `namespaceDeletionStatusTrackerKey(ns, name)` was meant to be more like `resourceDeletionStatusTrackerKey(resource, ns, name string) string` ?